crates/host/src/module.rs

```diff
@@ -20,17 +52,26 @@ impl ModuleCache {
         key: [u8; 32],
         wasm: &[u8],
     ) -> Result<Arc<Module>, WasmError> {
-        let store = Store::new(&Universal::new(Cranelift::default()).engine());
-        let module =
-            Module::from_binary(&store, wasm).map_err(|e| WasmError::Compile(e.to_string()))?;
-        self.0.insert(key, Arc::new(module));
-        self.get(key, wasm)
+        let module = SERIALIZED_MODULE_CACHE.write().get(key, wasm)?;
+        let arc = Arc::new(module);
+        self.0.insert(key, Arc::clone(&arc));
+        Ok(arc)
     }
 
     pub fn get(&mut self, key: [u8; 32], wasm: &[u8]) -> Result<Arc<Module>, WasmError> {
-        match self.0.get(&key) {
-            Some(module) => Ok(Arc::clone(module)),
-            None => self.get_with_build_cache(key, wasm),
+        let mut rng = rand::thread_rng();
+        let u: u8 = rng.gen();
```

Comment:
Wouldn't this be slower then just counting with an atomic size?

Reply:
yes. do we need a random number or just a unique number?